src/runtime/server/services/nuxtAuthHandler.ts

```diff
@@ -231,9 +231,14 @@ export const NuxtAuthHandler = (nuxtAuthOptions?: AuthOptions) => {
 }
 
 export const getServerSession = async (event: H3Event) => {
+  const authBasePath = useRuntimeConfig().public.auth.basePath
+  // avoid running auth middleware on auth middleware (see #186)
+  if (event.path.startsWith(authBasePath)) {
+    return
```

Comment:
let's also `return null` here so that the signature stays `null | Session`

Reply:
good point, updated